### DISCUSSION TITLE
Create 'loiter' mode that uses GPS to hold position along flight path

### DISCUSSION THREAD

**Discussion by pnduffy** [iNavFlight/inav#7867] [Ideas]
I have several copters and planes that use ArduPilot, which has a 'loiter' mode that uses GPS to hold the current line of travel in the direction of vehicle movement.   The current 'PositionHold' only seems to hold when the sticks are released, otherwise the vehicle drifts in alt-hold mode.   Heading-hold uses the compass, not GPS, so drifting horizontally is possible.  It would be a nice feature to add this mode.

**Comment by MrD-RC** [comment]
To me, loiter or position hold are to keep the craft in the current position. Not for moving. 

It sound like you want a version of fixed wing Cruise mode for multirotors.

**Reply by avsaase** [reply]
I made an issue about that a while ago: https://github.com/iNavFlight/inav/issues/6843

**Reply by pnduffy** [reply]
MrD-RC,  I haven't tried cruise mode with planes, but if it uses GPS to hold the commanded direction, then yes.  It's simply the same idea as a waypoint navigation, where the copter flies in a line only using the sticks to control the speed and initial direction.   ArduCopter has this mode and it's a nice feature.
